packages/amplify-graphql-api-construct-tests/src/sql-tests-common/sql-array-objects.ts

```diff
@@ -0,0 +1,255 @@
+import * as path from 'path';
+import * as fs from 'fs-extra';
+import { LambdaClient, GetProvisionedConcurrencyConfigCommand } from '@aws-sdk/client-lambda';
+import { ImportedRDSType } from '@aws-amplify/graphql-transformer-core';
+import AWSAppSyncClient, { AUTH_TYPE } from 'aws-appsync';
+import { createNewProjectDir, deleteProjectDir } from 'amplify-category-api-e2e-core';
+import { initCDKProject, cdkDeploy, cdkDestroy } from '../commands';
+import { SqlDatatabaseController } from '../sql-datatabase-controller';
+import { CRUDLHelper } from '../utils/sql-crudl-helper';
+import { contactFieldMap } from './schemas/sql-array-objects/field-map';
+import { ONE_MINUTE } from '../utils/duration-constants';
+
+export const testGraphQLAPIArrayAndObjects = (
+  options: {
+    projFolderName: string;
+    region: string;
+    connectionConfigName: string;
+    dbController: SqlDatatabaseController;
+    resourceNames: { sqlLambdaAliasName: string };
+  },
+  testBlockDescription: string,
+  engine: ImportedRDSType,
+): void => {
+  describe(`${testBlockDescription} - ${engine}`, () => {
+    let projRoot;
+    let region, lambdaFunctionName, lambdaAliasName;
+
+    let dbController: SqlDatatabaseController;
+    let contactTableCRUDLHelper: CRUDLHelper;
+
+    beforeAll(async () => {
+      ({
+        region,
+        dbController,
+        resourceNames: { sqlLambdaAliasName: lambdaAliasName },
+      } = options);
+      const { projFolderName, connectionConfigName } = options;
+
+      const templatePath = path.resolve(path.join(__dirname, '..', '__tests__', 'backends', 'sql-models'));
```

Comment:
My thoughts of using an universal `configurable-stack`:

- It's totally feasible to have all tests depends on a single stack with every test defines the specific parameter through various command `json` files, but more time-consuming than have individual CDK per test;
- This requires expansion of current configurable stack functionalities, including support of other authorization modes like OIDC, IAM, etc., and the shift to `AmplifyAuth` construct from individual `UserPool`. Plus, `AmplifyAuth` provides a convenient way of managing `auth`-related resources;
- Another way of supporting different resources requirements of tests is to through more helper functions instead of a centralized provider like `AmplifyAuth` based on command files parameters, e.g. when the test is about IAM, then a helper called `createIAMIdentityPool` is applied to provision the resource;
- Reducing the number of CDK related files to one stack will benefit maintaining the fixtures, and may need breaking and larger scale changes in current `configurable-stack`, which may lead to redesign of some of the tests' resources provision.

I could try to utilize the `configurable-stack` to provision the resources, and figure if OIDC could be supported as well for better management over the test fixtures.

Reply:
Thanks for your thoughts on this. For now, let's defer any move to configurable-stack, and we can consider a holistic design later.